src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java

```diff
@@ -481,7 +482,13 @@ private ByteBuffer current() throws IOException {
                     if (debug.on()) debug.log("Next Buffer");
                     currentBuffer = currentListItr.next();
                 } catch (InterruptedException ex) {
-                    // continue
+                    try {
+                        close();
+                    } catch (IOException ignored) {
+                    }
+                    // Throw InterruptedIOException where the initCause is
+                    // set to the caught InterruptedException
+                    throw Utils.toInterruptedIOException(ex);
```

Comment:
Don't you think you need to  restores the interrupt status of the thread by invoking the Thread.currentThread().interrupt() method and allowing the caller to detect the interrupt if it needs to ?

Reply:
By convention, any method that exits by throwing an InterruptedException clears interrupt status when it does so. Since we're throwing InterruptedIOException here, I don't think we need to set the interrupt flag. But the caller might do so if they want.

Reply:
Hmmm... Maybe a better solution would be to throw "ClosedByInterruptException" since we're also closing the input stream. And the specification of  "ClosedByInterruptException"  clearly state that the interrupt status of the thread will be preserved. Good question @vyommani .

Reply:
I am not 100% sure but for me the best option is call close() and  just restore the interrupt status let the caller decide how to handle the interrupt.

Reply:
InterruptedIOException is not equivalent to InterrupedException in this way AFAIK. The thread interrupt status should be reset if _any_ exception other than InterruptedException is thrown AFAIK, otherwise thread interruption is likely to be lost.